### DISCUSSION TITLE
emulate laststatus=3 with laststatus=0 and without the global line

### DISCUSSION THREAD

**Discussion by tnxz** [neovim/neovim#29510] [Q&A]
```
vim.o.ls = 0
vim.o.stl = "%{repeat('─',winwidth('.'))}"
```

In the `fillchars` docs
```
Note that "horiz", "horizup", "horizdown", "vertleft", "vertright" and
"verthoriz" are only used when 'laststatus' is 3, since only vertical
window separators are used otherwise.
```

so `vim.o.ls = 0` cannot properly simulate `vim.o.ls = 3`

does any one have other ideas

**Comment by Shougo** [comment]
What is the simulate?
I don't know what you want to do.

**Comment by tnxz** [comment]
simluate `vim.o.laststatus=3` without the global statusline

**Reply by Shougo** [reply]
I don't understand.
You want to remove statusline when split windows?  It cannot.
Because statusline is window separator.

If you want to hide the statusline.

```vim
set statusline=%{'─'->repeat(winwidth(0))}
set laststatus=0
```

It seems laststatus=3 behavior.

**Reply by Shougo** [reply]
And I use the highlights.

```vim
if &laststatus > 0
  highlight StatusLine    gui=bold guibg=#c8c8d8 guifg=#151505
  highlight StatusLineNC  gui=reverse guibg=#c8c8d8 guifg=#151505
else
  " Disable StatusLine highlights
  highlight StatusLine    gui=NONE guibg=#1c1c1c guifg=#aaaaaa
  highlight StatusLineNC  gui=NONE guibg=#1c1c1c guifg=#1c1c1c
endif
```

**Reply by tnxz** [reply]
1. when 

```
set statusline=%{'─'->repeat(winwidth(0))}
set laststatus=0
```
<img width="1680" alt="ls=0" src="https://github.com/neovim/neovim/assets/91027295/bcb7b972-efbd-4232-be3a-2b0f35d91ace">

2. when `ls=3`

<img width="1680" alt="ls=3" src="https://github.com/neovim/neovim/assets/91027295/d16f3722-1bef-46c6-b287-ad6dfc2813ee">

here when using (1) everything looks good except the characters at the intersection missing which i cannot define as they are only available when `ls=3` as mentioned in the docs.

**Reply by tnxz** [reply]
I want to be able to define these characters `horiz`, `horizup`, `horizdown`, `vertleft`, `vertright` and
`verthoriz` with `ls=0` which is not possible

**Reply by Shougo** [reply]
Ah, I get it.  So it should be feature request.

It is `fillchars` statusline support when `laststatus!=3`.

**Reply by Shougo** [reply]
Hm....  I have read the code and the implementation is hard.

**Reply by Shougo** [reply]
Well.  You can use the config.

```vim
autocmd WinEnter,CursorHold *
      \ : if winlayout()[1]->type() != v:t_list
      \ |   set laststatus=0
      \ | else
      \ |   set laststatus=3
      \ | endif
```

**Reply by Shougo** [reply]
You can emulate by the highlight.

```vim
highlight StatusLine    gui=underline guibg=#2c2c2c guifg=#c8c8d8
highlight StatusLineNC  gui=underline guibg=#2c2c2c guifg=#c8c8d8
highlight WinSeparator gui=NONE guibg=#2c2c2c guifg=#c8c8d8
```

**Comment by SirWrexes** [comment]
[colofrul-winseps.nvim](https://github.com/nvim-zh/colorful-winsep.nvim) sorta emulates the behaviour you're looking for, with the sweet bonus of making you aware which split is active.

**Comment by sleeptightAnsiC** [comment]
(related to https://github.com/neovim/neovim/issues/22714 )
